autoware_adapi_v1_msgs/perception/msg/ObjectKinematics.msg

```diff
@@ -0,0 +1,5 @@
+geometry_msgs/PoseWithCovariance pose
+geometry_msgs/TwistWithCovariance twist
+geometry_msgs/AccelWithCovariance accel
```

Comment:
I think these `XWithCovariance` messages are Too large.

For example [Pose](https://github.com/ros2/common_interfaces/blob/rolling/geometry_msgs/msg/Pose.msg) message contains a Point and a Quaternion which has 7 double values in total. But [Covariance](https://github.com/ros2/common_interfaces/blob/rolling/geometry_msgs/msg/PoseWithCovariance.msg) part adds 36 double values extra to the message.

And we generally don't need covariance of this size. We generally just have a variance in the 6 fields which make up the diagonal of the 6x6 matrix.

@isamu-takagi @yukkysaito @miursh what do you think? Should we create a new messages in `XWithVariance` format which have smaller footprint? And maybe even with variance part with 6 `float32`s.

Right now we are using [autoware_auto_perception_msgs/msg/DetectedObject.idl](https://github.com/tier4/autoware_auto_msgs/blob/tier4/main/autoware_auto_perception_msgs/msg/DetectedObject.idl) which add these pose, twist, accel with covariances which could be 7(xyz xyzw) + 6(xyzrpy errors) making 13 doubles per item. And for 3, it could be 13x3=39 doubles per Kinematics item. But right now it is (7+36)x3=129 doubles per item. 

And these messages with unnecessarily large memory footprints are passed around. And [when rviz2 reads them](https://github.com/autowarefoundation/autoware.universe/blob/main/common/autoware_auto_perception_rviz_plugin/src/object_detection/detected_objects_display.cpp#L49), I think it also causes lags with large object counts. I am also working on a benchmarking test node to prove it but it might take a while to work on it. https://www.youtube.com/watch?v=E86TWT03UXw here is a video from my initial attempts. I should probably create an issue to discuss lol.

Reply:
I'm with Fatih on this.
I would like to know why we want these fields to be included for AD API. What are the use-cases of having covariance. (Same goes with `accelearation` and some of the fields in other message like `existence_probability` and `confidence`, although I don't care too much about those compared to covariance because they are relatively small in size)

Reply:
Thank you for comment.
Please wait a moment as I am preparing a simple figure to explain my opiniton.

Reply:
>And maybe even with variance part with 6 float32s.

If we do not allow rotation of the variance in the object coordinates, as in the image here, then six is fine.
![image](https://user-images.githubusercontent.com/8327598/225544266-45f54b6a-9729-462a-97ee-95ed302719cb.png)

If the following expressions are to be allowed, 7 or 8 variance value will be required.
![image](https://user-images.githubusercontent.com/8327598/225544315-b2a84523-ef17-4584-9686-ebac8bdca7c1.png)


Honestly, I too think 36 arrays are useless.
However, XXXwithCovariance is a standard format for ROS and has the advantage of being able to use some tools, so I am torn.

I also think the following is good.  I think never use position-rotation covariance.
![image](https://user-images.githubusercontent.com/8327598/225552647-fee61d20-1dde-4929-bc94-655c7eebce27.png)

Reply:
We had some further discussion between @yukkysaito, @xmfcx and me.
At least we agree that covariance is unnecessary for AD API at all unless there is external software that needs covariance .

(Some off topic discussion)
We have also discussed about what we should do with [PredictedObjectKinetmatics](https://github.com/tier4/autoware_auto_msgs/blob/tier4/main/autoware_auto_perception_msgs/msg/PredictedObjectKinematics.idl) which is used inside Autoware. Our conclusion was that we would probably want to remove covariance for velocity and acceleration, and we could also remove unnecessary fields of covariance when we port the message to autoware_msgs. @xmfcx will create a discussion to discuss in further detail on this.

Reply:
Alright, thanks a lot @yukkysaito 
I will create an issue and a sample message in the https://github.com/autowarefoundation/autoware_msgs to replace [autoware_auto_perception_msgs/msg/DetectedObject.idl](https://github.com/tier4/autoware_auto_msgs/blob/tier4/main/autoware_auto_perception_msgs/msg/DetectedObject.idl) with something simpler.

And for this PR, we can make it even simpler by removing covariances altogether. Because this is for visualization from the PR title.

And for the autoware_msgs, I have following covariance representations in my head:
Add a new message for covariance:
```
float32 var_pos_x
float32 var_pos_y
float32 var_pos_z
float32 var_rot_r
float32 var_rot_p
float32 var_rot_y
float32 cov_pos_xy
float32 cov_pos_xz
float32 cov_pos_yz
```

@yukkysaito what do you think?

> However, XXXwithCovariance is a standard format for ROS and has the advantage of being able to use some tools, so I am torn.

Yes, the ROS standard is to use `XWithCovariance` but do you have any libraries or visualization tools where this change would affect us negatively?

Most tools/libraries use Pose, Orientation parts from what I remember.

Reply:
Also https://en.wikipedia.org/wiki/Covariance_matrix says:
> Any [covariance](https://en.wikipedia.org/wiki/Covariance) matrix is [symmetric](https://en.wikipedia.org/wiki/Symmetric_matrix) and [positive semi-definite](https://en.wikipedia.org/wiki/Positive_semi-definite_matrix) and its main diagonal contains [variances](https://en.wikipedia.org/wiki/Variance) (i.e., the covariance of each element with itself).

So half of the information is redundant/unnecessary, at worst, we only need half of the matrix anyway.

Reply:
@xmfcx @yukkysaito @mitsudome-r 
Thank you for your commend. Yes, I do agree for the adapi, we do not need to covariance as it will make the msg big.

> At least we agree that covariance is unnecessary for AD API at all unless there is external software that needs covariance .
I will confirm with one of the external software that going to use this api.